ext/drreg/drreg.c

```diff
@@ -410,7 +410,12 @@ drreg_insert_restore_all(void *drcontext, instrlist_t *bb, instr_t *inst,
     uint aflags = (uint)(ptr_uint_t)drvector_get_entry(&pt->aflags.live, pt->live_idx);
     if (!pt->aflags.native &&
         (force_restore ||
-         TESTANY(EFLAGS_READ_ARITH, instr_get_eflags(inst, DR_QUERY_DEFAULT)) ||
+         /* On AArch64, unrecognized instructions are decoded as instances of a generic
+          * instruction, OP_xx. For these instructions conservatively restore aflags in
+          * case they read/write the aflags.
+          */
+         IF_AARCH64(instr_get_opcode(inst) == OP_xx ||)
```

Comment:
Given that all of the missing opcodes are vector instructions, I would expect a very small fraction to read or write flags (or GPR's for that matter).  I wonder if adding the couple of opcodes that affect flags and GPR's would not be much work.  Though it's certainly more work than this workaround.  Let's wait for advice from those more familiar with the a64 decoder: @AssadHashmi?

Reply:
> Given that all of the missing opcodes are vector instructions, I would expect a very small fraction to read or write flags (or GPR's for that matter). I wonder if adding the couple of opcodes that affect flags and GPR's would not be much work. Though it's certainly more work than this workaround. Let's wait for advice from those more familiar with the a64 decoder: @AssadHashmi?

The resourcing planned to address AArch64 decoder gaps has been diverted elsewhere at the present time. Reluctantly, I'd advise carrying on with the workaround. This is far from ideal for AArch64 support but we currently have to work within limited constraints. Hopefully the situation will change later in the year.

Reply:
Thinking more about the implications here: workarounds will have to added to multiple places and not just drreg.  The drcachesim tracer does GPR analysis to elide same-base addresses.  The memval tool looks at register usage.  Dr. Memory, being ported now, looks at everything.  (The core does flags analysis for fragment prefixes and stay-on-trace comparisons -- though I believe these have no problem today due to flags-less IBL and no trace support yet for AArch64.)

@AssadHashmi is it the case that OP_xx insructions have GPR operands in place as hinted at here https://github.com/DynamoRIO/dynamorio/issues/2626#issuecomment-351976969?  Does that reliably cover all GPR usage in such instructions?

Reply:
> @AssadHashmi is it the case that `OP_xx` instructions have GPR operands in place as hinted at here #2626 (comment)?

@derekbruening yes I believe that is the case. @egrimley can confirm.

> Does that reliably cover all GPR usage in such instructions?

The majority, if not all, `OP_xx` instructions are SIMD. Of those, GPR usage is restricted to an index register. Based on that you should be reliably covered. There may be cases of non-SIMD `OP_xx` but even then you should be ok.